### PR TITLE
Added SPI write and transfer functions

### PR DESCRIPTION
**Pull request by ioanatheodora**
Added SPI module from `stm32f7xx_hal` crate. Implemented write and transfer functions for writing and reading bytes on SPI3 peripheral.

### CODE REVIEW

File: firmware/obc/cubesat-1-fw-obc/src/board_demo.rs [hunk 1]
```diff
@@ -1,538 +1,588 @@
...
+
+        // Enable the button interrupt
+        unsafe {
+            NVIC::unmask::<interrupt>(interrupt::EXTI15_10);
+        }
+
+        // Copy lot number slice from UID
+        let mut lot_num: [u8; 7] = Default::default();
+        lot_num.clone_from_slice(Uid::get().lot_num().as_bytes());
+
```
Comment (Andrei-Basarab) [inline firmware/obc/cubesat-1-fw-obc/src/board_demo.rs]: Add comment for this code block

File: firmware/obc/cubesat-1-fw-obc/src/board_demo.rs [hunk 2]
```diff
@@ -1,538 +1,588 @@
...
+   
+        let mut ncs = gpioc.pc9.into_push_pull_output();
+        let sck = gpioc.pc10.into_alternate();
+        let miso = gpioc.pc11.into_alternate();
+        let mosi = gpioc.pc12.into_alternate();
+        
+        // Set NCS pin to high (disabled) initially
+        ncs.set_high();
+
+        // Initialize SPI
```
Comment (Andrei-Basarab) [inline firmware/obc/cubesat-1-fw-obc/src/board_demo.rs]:
```suggestion
        // Initialize SPI3

```

File: firmware/obc/cubesat-1-fw-obc/src/board_demo.rs [hunk 3]
```diff
@@ -1,538 +1,588 @@
...
+                cal110: VtempCal110::get().read(),
+                timer: TimerUs::default(),
+            },
+            serial: BoardDemoSerial {
+                tx: serial_tx,
+                rx: serial_rx,
+            }, 
+            spi: BoardDemoSpi {
+                spi,
+                ncs,
```
Comment (Andrei-Basarab) [inline firmware/obc/cubesat-1-fw-obc/src/board_demo.rs]: Rename `ncs` to `cs`

File: firmware/obc/cubesat-1-fw-obc/src/board_demo.rs [hunk 4]
```diff
@@ -1,538 +1,588 @@
...
+            _ => { /* Do nothing in Idle state */ }
+        }
+    }
+
+    pub fn spi_write(&mut self, buffer: &mut [u8]) -> Result<(), Error> {
+        self.spi.spi.write(buffer)
+    }
+
+
+    // read bytes function
```
Comment (Andrei-Basarab) [inline firmware/obc/cubesat-1-fw-obc/src/board_demo.rs]: Add doc comments with `///`

File: firmware/obc/cubesat-1-fw-obc/src/board_demo.rs (hunk 1 above)
Comment (ioanatheodora) [reply firmware/obc/cubesat-1-fw-obc/src/board_demo.rs]: Fixed!

**Review by Andrei-Basarab — APPROVED** [review]

**Pull request merged**